### PR TITLE
Fix: issue-637

### LINKED ISSUE #637: ps_mainmenu issue when too many items for one row
Prestashop 8.1.6.
Hummingbird 0.2.0.

Desktop issue.

The main menu is working correctly only if a few items are shown:
image

If there are more than fit one row, it gets broken:
image

Expected behavior:
Use more than one row if needed.

### PR DESCRIPTION
**Pull request by tblivet**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Allow the main menu to display on multiple rows instead of using the full row width when containing many items. Also, change the submenu top position to handle the case when multiple rows contain submenus.
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | 
| Fixed ticket?     | Fix https://github.com/PrestaShop/hummingbird/issues/637
| Sponsor company   | @PrestaShopCorp
| How to test?      | You can refer to the issue https://github.com/PrestaShop/hummingbird/issues/637

### CODE REVIEW

**Review by nicosomb — APPROVED** [review]
OK for the css / tpl part.

**Review by boherm — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @tblivet ,

Thanks for the PR !

Not all elements are displayed, but I guess that i'ts the wanted behavior, as 50 elements can't be displayed all in one line. ✅ 

I can't click on any submenu. :x:

https://github.com/user-attachments/assets/9a914ff7-c03e-4820-8976-45b9c2b67c82

Could you check ?
Thanks!

**Review by florine2623 — APPROVED** [review]
Hello @tblivet ,

LGTM on the shape of the menu ✅ 

https://github.com/user-attachments/assets/21de17f5-4156-4d31-b74e-9be333b491ab

When there are more than 3 sub-categories, the last one is not well displayed. But that's an issue for another PR :)
Screenshot 2024-07-19 at 09 38 08

It is QA ✅

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
